client/components/marketing-survey/cancel-purchase-form/isSurveyFilledIn.js

```diff
@@ -0,0 +1,18 @@
+export default function isSurveyFilledIn( survey ) {
+	const answeredBothQuestions = survey.questionOneRadio && survey.questionTwoRadio;
+	let needsText = false;
+
+	if ( ! answeredBothQuestions ) {
+		return false;
+	}
+
+	if ( survey.questionOneRadio === 'anotherReasonOne' && survey.questionOneText === '' ) {
+		needsText = true;
+	}
+
+	if ( survey.questionTwoRadio === 'anotherReasonTwo' && survey.questionTwoText === '' ) {
+		needsText = true;
+	}
+
+	return ! needsText;
+}
```

Comment:
it's curious why we need the `needsText` variable since all we're doing is returning and never resetting it. why not eliminate the variable and just `return` in each conditional?